pkg/distributor/distributor.go

```diff
@@ -1401,19 +1401,37 @@ func NextOrCleanup(next PushFunc, pushReq *Request) (_ PushFunc, maybeCleanup fu
 
 // Push is gRPC method registered as client.IngesterServer and distributor.DistributorServer.
 func (d *Distributor) Push(ctx context.Context, req *mimirpb.WriteRequest) (*mimirpb.WriteResponse, error) {
+	start := time.Now()
 	pushReq := NewParsedRequest(req)
 	pushReq.AddCleanup(func() {
 		mimirpb.ReuseSlice(req.Timeseries)
 	})
 
 	pushErr := d.PushWithMiddlewares(ctx, pushReq)
+
+	// TODO(gotjosh): Is this better as a middleware?
```

Comment:
I think this is probably correct that it would be better as a middleware.

Reply:
This was not up for review yet - it the end the middleware was the way to go for other reasons that I realise after testing 😄